### PR TITLE
feat(portal): Time based policies

### PR DESCRIPTION
**Pull request by AndrewDryga**
Flows authorized by time-based policies will now expire at the latest time permitted by the policy.

### CODE REVIEW

File: elixir/apps/domain/lib/domain/flows.ex [hunk 1]
```diff
@@ -32,7 +32,7 @@ defmodule Domain.Flows do
     with :ok <- Auth.ensure_has_permissions(subject, Authorizer.create_flows_permission()),
          {:ok, resource} <-
            Resources.fetch_and_authorize_resource_by_id(resource_id, subject, opts),
-         {:ok, policy} <- fetch_conforming_policy(resource, client) do
+         {:ok, policy, conformation_expires_at} <- fetch_conforming_policy(resource, client) do
```
Comment (jamilbk) [inline elixir/apps/domain/lib/domain/flows.ex:35]: TIL conformation is a word 😆

**Review by jamilbk — APPROVED** [review]
LGTM

File: elixir/apps/domain/lib/domain/policies/condition/evaluator.ex [hunk 2]
```diff
@@ -6,113 +6,183 @@ defmodule Domain.Policies.Condition.Evaluator do
...
         end)
 
       {:error, _reason} ->
-        false
+        nil
     end
   end
 
+  # Merge ranges, eg. 4-11,11-22 = 4-22
+  defp merge_joint_time_ranges(time_ranges) do
```
Comment (jamilbk) [inline elixir/apps/domain/lib/domain/policies/condition/evaluator.ex]: Would it be a good idea to mark this function public and have a few unit tests just for it?

Comment (AndrewDryga) [reply elixir/apps/domain/lib/domain/policies/condition/evaluator.ex]: @jamilbk we have a few cases that test this indirectly but I'll add a few unit tests directly for it too

**Pull request merged**